api/src/main/java/io/strimzi/api/kafka/model/status/KafkaConnectorStatus.java

```diff
@@ -22,7 +25,19 @@
 @JsonPropertyOrder({ "conditions", "observedGeneration", "url" })
 @EqualsAndHashCode
 @ToString(callSuper = true)
+@SuppressFBWarnings("SE_BAD_FIELD")
 public class KafkaConnectorStatus extends Status {
     private static final long serialVersionUID = 1L;
 
+    private JsonNode connector;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Description("The connector status, as reported by the Kafka Connect REST API.")
+    public JsonNode getConnector() {
```

Comment:
Originally I went to the trouble of writing classes to model the API response schema, but I decided that was probably a bad idea since we end up coupling our CRD schema to Connect's API response schema (which we don't control, thus creating compatibility problems for us if their response gains extra properties).

So I decided that our schema should basically have this property whose JSON type should be `object`. Usually we use a `Map<String, Object>` for that, I thought it wasn't trivial to get Jackson to do that. Looking again, maybe it's a lot easier than I thought... I'll take a look.

Reply:
Do you remember we had problem using JsonNode with the `KafkaClusterRebalanceStatus.optimizationResult` field so we moved to use `Map<String, Object>`. The optimization result is a JSON quite like the connector information you get from the Connect REST API.

Reply:
I don't recall the problem, but I was prepared to "fix" the `CrdGenerator` and `DocGenerator` to make `JsonNode` work. But having said that, I'm going to try again to make `Map` work.

Reply:
So I wonder if this is just shifting the Connect APi stability to the users when we do not model the classes. But I guess I can live with it.

Reply:
Indeed, but the alternative (us being in the loop to provide compatibility and/or needing to update a schema for it) seemed more attractive to me.